src/Microsoft.AspNetCore.Authorization.Policy/AuthenticationPolicy.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    /// <summary>
+    /// AuthenticationPolicy assigns a set of schemes for all of the authentication actions.
+    /// </summary>
+    public class AuthenticationPolicy
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="authenticateSchemes"></param>
+        /// <param name="challengeSchemes"></param>
+        /// <param name="forbidSchemes"></param>
+        public AuthenticationPolicy(IEnumerable<string> authenticateSchemes,
+            IEnumerable<string> challengeSchemes,
+            IEnumerable<string> forbidSchemes)
+        {
+            AuthenticateSchemes = authenticateSchemes ?? new string[0];
+            ChallengeSchemes = challengeSchemes ?? new string[0];
+            ForbidSchemes = forbidSchemes ?? new string[0];
+        }
+
+        /// <summary>
+        /// The authentication schemes that should be used to construct the ClaimsPrincipal during authentication.
+        /// </summary>
+        public IEnumerable<string> AuthenticateSchemes { get; }
+
+        /// <summary>
+        /// The authentication schemes that should be challenged. 
+        /// </summary>
+        public IEnumerable<string> ChallengeSchemes { get; }
```

Comment:
Should be one, no? Many challenge schemes is odd.

Reply:
Yeah its odd, I believe one of the original reasons why we ever supported this was an upgrade flow, where a user would be signed in, but they might require more permissions which would result in additional challenge, @Tratcher @blowdart do you guys remember the details of why we ever wanted multiple authentication schemes for authorization (other than bearer + cookies)

Reply:
Yea, the upgrade was definitely a scenario

Reply:
Challenging Basic, Bearer, Windows, on the same response works, they stack. That's what IIS does if you enable multiple auth types.

Reply:
I know what it does but it feels like a pit of failure. How often is this usable?

Reply:
What's the verdict here @blowdart @Tratcher @davidfowl 

Can we limit Challenge/Forbid to a single scheme for an AuthenticationPolicy?  Multiples would still be supported on the APIs I guess.